src/include/cpu/rpp_cpu_common.hpp

```diff
@@ -2081,7 +2081,101 @@ inline RppStatus custom_convolve_image_host(T* srcPtr, RppiSize srcSize, U* dstP
 
 
 
-// Compute Functions
+// Compute Functions for RPP Tensor API
+
+inline RppStatus compute_gridmask_masks_16_host(__m128 *pCol, __m128 *pGridRowRatio, __m128 pCosRatio, __m128 pSinRatio, __m128 pGridRatio, __m128 *pMask)
+{
+    __m128 pCalc[2];
+
+    pCalc[0] = _mm_fmadd_ps(pCol[0], pCosRatio, pGridRowRatio[0]);
+    pCalc[1] = _mm_fmadd_ps(pCol[0], pSinRatio, pGridRowRatio[1]);
+    pCalc[0] = _mm_cmpge_ps(_mm_sub_ps(pCalc[0], _mm_floor_ps(pCalc[0])), pGridRatio);
+    pCalc[1] = _mm_cmpge_ps(_mm_sub_ps(pCalc[1], _mm_floor_ps(pCalc[1])), pGridRatio);
+    pMask[0] = _mm_or_ps(pCalc[0], pCalc[1]);
+
+    pCalc[0] = _mm_fmadd_ps(pCol[1], pCosRatio, pGridRowRatio[0]);
+    pCalc[1] = _mm_fmadd_ps(pCol[1], pSinRatio, pGridRowRatio[1]);
+    pCalc[0] = _mm_cmpge_ps(_mm_sub_ps(pCalc[0], _mm_floor_ps(pCalc[0])), pGridRatio);
+    pCalc[1] = _mm_cmpge_ps(_mm_sub_ps(pCalc[1], _mm_floor_ps(pCalc[1])), pGridRatio);
+    pMask[1] = _mm_or_ps(pCalc[0], pCalc[1]);
+
+    pCalc[0] = _mm_fmadd_ps(pCol[2], pCosRatio, pGridRowRatio[0]);
+    pCalc[1] = _mm_fmadd_ps(pCol[2], pSinRatio, pGridRowRatio[1]);
+    pCalc[0] = _mm_cmpge_ps(_mm_sub_ps(pCalc[0], _mm_floor_ps(pCalc[0])), pGridRatio);
+    pCalc[1] = _mm_cmpge_ps(_mm_sub_ps(pCalc[1], _mm_floor_ps(pCalc[1])), pGridRatio);
+    pMask[2] = _mm_or_ps(pCalc[0], pCalc[1]);
+
+    pCalc[0] = _mm_fmadd_ps(pCol[3], pCosRatio, pGridRowRatio[0]);
+    pCalc[1] = _mm_fmadd_ps(pCol[3], pSinRatio, pGridRowRatio[1]);
+    pCalc[0] = _mm_cmpge_ps(_mm_sub_ps(pCalc[0], _mm_floor_ps(pCalc[0])), pGridRatio);
+    pCalc[1] = _mm_cmpge_ps(_mm_sub_ps(pCalc[1], _mm_floor_ps(pCalc[1])), pGridRatio);
+    pMask[3] = _mm_or_ps(pCalc[0], pCalc[1]);
+
+    pCol[0] = _mm_add_ps(pCol[0], xmm_p16);
+    pCol[1] = _mm_add_ps(pCol[1], xmm_p16);
+    pCol[2] = _mm_add_ps(pCol[2], xmm_p16);
+    pCol[3] = _mm_add_ps(pCol[3], xmm_p16);
+
+    return RPP_SUCCESS;
+}
+
+inline RppStatus compute_gridmask_masks_4_host(__m128 &pCol, __m128 *pGridRowRatio, __m128 pCosRatio, __m128 pSinRatio, __m128 pGridRatio, __m128 &pMask)
+{
+    __m128 pCalc[2];
+
+    pCalc[0] = _mm_fmadd_ps(pCol, pCosRatio, pGridRowRatio[0]);
+    pCalc[1] = _mm_fmadd_ps(pCol, pSinRatio, pGridRowRatio[1]);
+    pCalc[0] = _mm_cmpge_ps(_mm_sub_ps(pCalc[0], _mm_floor_ps(pCalc[0])), pGridRatio);
+    pCalc[1] = _mm_cmpge_ps(_mm_sub_ps(pCalc[1], _mm_floor_ps(pCalc[1])), pGridRatio);
+    pMask = _mm_or_ps(pCalc[0], pCalc[1]);
+    pCol = _mm_add_ps(pCol, xmm_p4);
+
+    return RPP_SUCCESS;
+}
+
+inline RppStatus compute_gridmask_result_48_host(__m128 *p, __m128 *pMask)
+{
+    p[0] = _mm_and_ps(p[0], pMask[0]);
```

Comment:
better to do in AVX

Reply:
A lot of the host implementation for gridmask was before the avx discussion. I'll probably issue a separate PR to add the AVX version since SSE was already implemented. Wanted to merge this soon since the ricap implementations were dependent on the merge to avoid conflicts.